kornia/feature/affine_shape.py

```diff
@@ -0,0 +1,118 @@
+from typing import Tuple
+
+import torch
+import torch.nn as nn
+import math
+from kornia.filters import get_gaussian_kernel2d
+from kornia.filters import SpatialGradient
+from kornia.feature.laf import (ellipse_to_laf,
+                                get_laf_scale,
+                                raise_error_if_laf_is_not_valid,
+                                scale_laf, make_upright)
+from kornia.feature import extract_patches_from_pyramid
+
+
+class PatchAffineShapeEstimator(nn.Module):
+    def __init__(self, patch_size: int = 19, eps: float = 1e-10):
+        """Module, which estimates the second moment matrix of the patch gradients in order to determine the
+        affine shape of the local feature as in
+        A. Baumberg (2000). "Reliable feature matching across widely separated views".
+        In Proc.of CVPR. pp. I:1774--1781.
+        Args:
+            patch_size: int, default = 19
+            eps: float, for safe division, default is 1e-10"""
```

Comment:
the docstrings diverges from the others. I'm fine since you checked the docs are well generated.

Reply:
This works, if add to rst:
.. autoclass:: PatchAffineShapeEstimator
   :members: forward

So 2 methods will be documented. I am doing it where forward() signature is different from standard single input-single output